hoodie-client/src/main/java/com/uber/hoodie/index/hbase/HBaseIndex.java

```diff
@@ -161,24 +209,43 @@ private Connection getHBaseConnection() {
       }
       return taggedRecords.iterator();
     }
+
+    private Get generateStatement(String key) throws IOException {
+      return new Get(Bytes.toBytes(key)).setMaxVersions(1)
+          .addColumn(SYSTEM_COLUMN_FAMILY, COMMIT_TS_COLUMN)
+          .addColumn(SYSTEM_COLUMN_FAMILY, FILE_NAME_COLUMN)
+          .addColumn(SYSTEM_COLUMN_FAMILY, PARTITION_PATH_COLUMN);
+    }
+
+    private boolean checkIfValidCommit(String commitTs) {
+      HoodieTimeline commitTimeline = hoodieTable.getCompletedCommitTimeline();
+      // Check if the last commit ts for this row is 1) present in the timeline or
+      // 2) is less than the first commit ts in the timeline
+      return !commitTimeline.empty() && (commitTimeline.containsInstant(
+          new HoodieInstant(false, HoodieTimeline.COMMIT_ACTION, commitTs)) ||
+          HoodieTimeline.compareTimestamps(commitTimeline.firstInstant().get().getTimestamp(),
+              commitTs, HoodieTimeline.GREATER));
+    }
   }
 
   @Override
   public JavaRDD<HoodieRecord<T>> tagLocation(JavaRDD<HoodieRecord<T>> recordRDD,
-      HoodieTable<T> hoodieTable) {
+                                              HoodieTable<T> hoodieTable) {
     return recordRDD.mapPartitionsWithIndex(this.new LocationTagFunction(hoodieTable), true);
   }
 
-  class UpdateLocationTask implements
-      Function2<Integer, Iterator<WriteStatus>, Iterator<WriteStatus>> {
+  class UpdateLocationTask implements Function2<Integer, Iterator<WriteStatus>, Iterator<WriteStatus>> {
```

Comment:
this is some old code, before I was fluent in java8 closures :). can we save some code here by using the functional api instead of extending `Function2`